### ISSUE TITLE
Bijin: Warmaidens SE, Wives SE, NPCS SE

### ISSUE THREAD

**Issue by jumbod999** [tiltedphoques/Mod-Compatibility#111]
# Mod information

#### Download link
https://www.nexusmods.com/skyrimspecialedition/mods/1825
https://www.nexusmods.com/skyrimspecialedition/mods/11247
https://www.nexusmods.com/skyrimspecialedition/mods/11287

#### Dependencies
- SKSE
  -  http://skse.silverlock.org/
  -  (https://github.com/tiltedphoques/Mod-Compatibility/issues/2)

# Compatibility

> Mod is NOT compatible, requires the skse64_loader.exe to be run, which SkyrimTogether.exe cannot run.  

# Additional Information

> Trying to run skse64_loader with SkyrimTogether,exe results in inability to locate Address Library for SKSE Plugins. 

> Running game through SkyrimTogether.exe properly will display characters with potato faces and purple hair (missing texture). Seems to be a problem with FaceGen.

**Comment by Astral3** [comment]
Skyrim Together automatically runs SKSE if properly installed. Attempting to link using the executable is unnecessary and demonstrates you likely did not follow instructions.

Considering that these are graphic mods, I would like a second opinion before this compatibility report is considered valid.

**Comment by jumbod999** [comment]
@Astral3 
SKSE64 is properly installed with

- skse64 Data folder
- skse64 src folder
- skse_1_6_353.dll
- skse64_loader.exe
- skse64_steam_loader.dll
All in the game directory.

Launching SkyrimTogether.exe through MO2 and making sure that SkyrimSE.exe was properly will allow some SKSE mods to work that I have tested (SkyUi, CBBE) so far seem to work correctly. However bijin does not as explained.

When launching the game through skse64_loader.exe Bijin works as intended and all characters have been changed.

My guess is that there is a problem with FaceGen judging by how the characters look (mushing potato face and purple texture hair).

**Comment by FedUpWizard** [comment]
I had issues with Modpocalypse npcs  https://www.nexusmods.com/skyrimspecialedition/mods/54422 , having neck seams and launching skse didnt have the issue, also read about it being a FaceGen issue, I then found this mod (Face Discoloration fix) https://www.nexusmods.com/skyrimspecialedition/mods/42441 and it fixed all the neck seams from STR. This May or May not fix your issue, Worth a try :)

**Comment by jumbod999** [comment]
@FedUpWizard 
Thanks for your response! Your solution half worked. 

When I load a save from inside a cell containing an effected character it works! 

However, if I move into another cell with an effected character(s) it doesn't work. Very weird issue.

**Comment by FedUpWizard** [comment]
Yeah wierd, I did see the devs chatting about in in #coding so hopefully it gets fixed. If you want, you can try modpocalypse its pretty good, causes brown face bugs, so be warned the author has a script somewhere to fix the bug though plus I hear, Easy NPC can also fix them.

**Comment by jumbod999** [comment]
@FedUpWizard 
Modpocalypse NPCS requires the Unofficial Patch correct? So i would need to remove the 4 free CC addons that came with the update?

Also, do you have a link to the #coding so I could see what the devs said? Thanks.

**Comment by FedUpWizard** [comment]
Its on discord, I think about from here I read https://discord.com/channels/247835175860305931/717070423254368286/997522316676305051 Yes it does require USSEP This is the pre CC content update for SSE 1.6/AE ver l Found https://www.nexusmods.com/skyrimspecialedition/mods/266?tab=files&file_id=241103

**Comment by Astral3** [comment]
Thank you for the additional info, and for some additional info from a certain Gandalf. A rather ...odd issue, but thanks for your report!

**Comment by Calhoun266** [comment]
From my testing, loading a new game through normal SKSE will generate the expected facegen. Loading this save in STR will then keep the facegen, however you must load **every single NPC** affected by the mods through normal SKSE or you will get potato faces.

Edit: it seems about 8/10 times changing zones will break the facegen unfortunately.

**Comment by Astral3** [comment]
I've tested this mod in the test build of the upcoming STR patch.
#### Additional mods:
 - [Face Discoloration Fix](https://www.nexusmods.com/skyrimspecialedition/mods/42441)

I did not note any particular issues which you described - no NPCs had potato faces, purple textural issues, or neck seam problems. The mod will be considered compatible until someone reports otherwise.

_A few NPCs did lack a face and only have floating teeth, but I believe that was due to my making a mistake when overwriting textures - especially as it wasn't reported previously._

**Comment by theslenderloris** [comment]
tested this mod with the most recent Skyrim update/SKSE/Skyrim Together (Sept 24th) and Bijin NPC's have purple hair and potato faces with a neck seam after reloading a cell. The mod will work if you load the game into a cell with a Bijin NPC but upon loading a new cell the purple hair and potato faces return.

used with and without Face Discoloration Fix, no change. 

edit (Sept 25th): so trying this on my Desktop instead of my Steam Deck had better results but going in and out of cells eventually broke the facegen. certain NPCs like Ysolda kept their appearance for a few loads but became purple haired and potato faced after a few cell changes.

**Comment by IKE200** [comment]
Did anyone ever find a working solution for this problem?

I did some testing myself and it seems to be comming donw to STR not loading FaceGeom data properly. I tested it with Seranaholic: https://www.nexusmods.com/skyrimspecialedition/mods/13027
When loading this mod just with basic Skyrim AE it works fine. it also works fine with launching through SKSE. When just STR is installed following the installation instructions (I verified it worked by connecting with a friend) Seranas face is missing completely and I just have floating teeth and purple hair.

I was able to reproduce this in default Skyrim and SKSE by deleting the FaceGeom nif under meshes\actors\character\FaceGenData\FaceGeom\Dawnguard.esm\00002b6c.nif from the Seranaholic mod.

When this file is missing it looks the same as it does when launching STR but with the FaceGeom in place.

A few observations I have made comparing it to changes from NPC mods that have been working for me. I compared it to Diversity: https://www.nexusmods.com/skyrimspecialedition/mods/5291

Seranaholic seems to accomplish its changes by applying multiple custom headparts to Serana in its esp. These headparts are responsible for her Hair, Head, Brows and Eyes. The custom parts Head and Eyes both then reference .nif files (actors\character\Serana\femalehead.nif and actors\character\Serana\eyesfemale.nif), which are not bundled into the mod. My guess is, that Skyrim can use the FaceGeom nifs to directly display the correct head model, without needing to use the basic models for head, hair, eyes etc. But STR for some reason prevents these nifs from being loaded. Skyrim then tries to load the meshes referenced in the headparts directly but if they are not present it then just displayes nothing.

I was able to reproduce this in diversity by deleting Seranas FaceGeom nif (actors\character\FaceGenData\FaceGeom\Dawnguard.esm\00002b6c.nif) from the mod and then also deleting her hair model (KS Hairdo's\Cancia.nif). In default Skyrim and SKSE this resulted in her Hair not being displayed. However as long as one of the two files were present the Hair would display correctly. In STR when just the FaceGeom was missing it displayed correctly but with just the Hair mesh missing the Hair was not displayed.

Both mods also reference .tri files for morph data. My guess is, that these files would also be needed to get the exact look as was originally intended by the mod author but they seem to be not needed for the Hair, face, eyes, etc. to atleast be displayed.

TLDR:
NPC overhauls are tricky. If the overhauls just include the FaceGeom nif files and not the nifs for hair, eyes, face etc. needed to generate the FaceGeom data the NPCs will not be displayed correctly. If the mod however includes all relevant nifs (maybe also tri files) the mod seems to work fine.
